backend/src/test/java/com/google/models/AnomalyTest.java

```diff
@@ -135,6 +136,19 @@ public void createAnomalyFromEmbeddedEntity_correctEmbeddedEntityToAnomalyConver
     assertEquals(convertedAnomaly, ANOMALY);
   }
 
+  @Test
+  public void createAnomalyFromEmbeddedEntity_dataPointsInChronologicalOrder() {
+    EmbeddedEntity anomalyEmbeddedEntity = ANOMALY.toEmbeddedEntity();
+    Anomaly convertedAnomaly = Anomaly.createAnomalyFromEmbeddedEntity(anomalyEmbeddedEntity);
+
+    // DATA_POINTS is already in sorted order by timestamp, so now line below is comparing 
+    // if the DATA_POINTS after entity -> anomaly conversion still has the points in order.
+    assertEquals(
+      DATA_POINTS.keySet().stream().collect(Collectors.toList()), 
```

Comment:
nit/optional: you can also rename DATA_POINTS --> SORTED_DATA_POINTS for future reference to ensure that if DATA_POINTS gets changed, it'll be less likely to get changed in a way that's not sorted.

for example, adding a new data point to the constant.

Reply:
i see, that makes sense :)
i will rename it!